cloudinit/config/cc_lxd.py

```diff
@@ -285,6 +285,7 @@ def maybe_cleanup_default(net_name, did_init, create, attach,
     if create:
         msg = "Deletion of lxd network '%s' %s"
         try:
+            _lxc(["network", "detach-profile", net_name, profile])
             _lxc(["network", "delete", net_name])
```

Comment:
So as this is, if I were looking at a log, I would not know which if these 2 commands failed.
The log would just show "deletion of lxd network 'xx' xx. Assuming it did not exist".

What we need is to try the `detach-profile`.  if that fails, we can "assume it did not exist".  But if it succeeds, then we *know* that it existed, so we can't ignore a failure of `delete`, so that failure should then be raised.

Reply:
Thanks; should be handled better now,